### ISSUE TITLE
Remove redundant `aria-disabled` in customize columns items

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#4797]
Reported by @kristastarr in the Datagrid [customize columns review](https://github.ibm.com/CDAI-design/pal/issues/1576) from the accessibility audit.

Flagged as a **recommendation**.

> **The ARIA attribute "aria-disabled" is redundant with the HTML attribute `"disabled"`**
> An ARIA attribute should not be used when there is a corresponding HTML attribute
> 
> ### What to do
> - Remove the unnecessary ARIA attribute that is redundant with HTML attribute.
> - Review the [Requirements for use of ARIA attributes in place of equivalent HTML attributes](https://www.w3.org/TR/html-aria/#docconformance-attr) table to help determine when an ARIA attribute should be correctly used.
>
> Rule ID: aria_attribute_redundant
> Reason ID: fail_redundant

![image](https://github.com/carbon-design-system/ibm-products/assets/1717398/7142db3e-f38e-47b5-b688-7362194724b7)
